test/coverband/collectors/coverage_test.rb

```diff
@@ -82,6 +82,15 @@ def teardown
     assert_equal false, @coverband.send(:track_file?, heroku_build_file)
   end
 
+  # verifies a fix where we were storing, merging, and tracking ignored files
+  # then just filtering them out of the final report
+  test 'ignores uses regex same as reporter does' do
+    regex_file = Coverband.configuration.current_root + '/config/initializers/fake.rb'
+    assert_equal true, @coverband.send(:track_file?, regex_file)
```

Comment:
I think if possible we should try and avoid calling private methods and instance vars from our tests. Can make refactoring difficult later.

Reply:
I know there is some advice about not testing private methods, I think we discussed it in another previous PR. I disagree with that testing advice as it makes tests far more complicated and makes it harder to interpret results. I think we will have different styles on this level of testing, which is cool as any project that has more than one dev has some divergent patterns.

Being able to target a highly specific method where complexity resides has at least for me, shown good value vs the tradeoffs of having to refactor tests when they change.

Reply:
Makes sense. Apologize for bringing the same issue again. Sometimes I think needing to test a private method, might be a good reason to extract the complicated logic into a class of its own.

Reply:
ha no worries, this seems to be one of the few things we have different coding opinions on ;)